### DISCUSSION TITLE
Okd on openstack through ipi method

### DISCUSSION THREAD

**Discussion by giteshchavan09** [okd-project/okd#1241] [Q&A]
Hi team,

We are Installing openshift cluster on OpenStack in a restricted network. but its failing. kindly help us to resolve the issue . 

 1. we have created image registry server successfully,
2. Creating the RHCOS image for restricted network installations
3. Enabling access with floating IP addresses
4. Image cache

We are able to create the manifeast file but its failing with create cluster,

Please find below install-config file for reference,

cat install-config.yaml
apiVersion: v1
baseDomain: poc.com
controlPlane:
  name: master
  platform:
    openstack:
      type: custom_EPH
  replicas: 3
compute:
- name: worker
  platform:
    openstack:
      type: custom_EPH
  replicas: 1
metadata:
  name: ipi-openstack
networking:
  clusterNetwork:
  - cidr: 10.244.0.0/16
    hostPrefix: 23
  machineNetwork:
  - cidr: 10.0.0.0/16
  networkType: OpenShiftSDN
  serviceNetwork:
  - 172.30.0.0/16
platform:
  openstack:
    region: region1
    cloud: openstack
    externalNetwork: public
    computeFlavor: custom_EPH
    apiFloatingIP: 10.157.36.21
    ingressFloatingIP: 10.157.37.20
    clusterOSImage: rhcos-4.10.3

Error

DEBUG Built from commit 3b701903d96b6375f6c3852a02b4b70fea01d694

INFO Waiting up to 20m0s (until 7:07PM) for the Kubernetes API at https://api.ipi-openstack.poc.com:6443...

DEBUG Still waiting for the Kubernetes API: Get "https://api.ipi-openstack.poc.com:6443/version": dial tcp 10.157.36.21:6443: i/o timeout

DEBUG Still waiting for the Kubernetes API: Get "https://api.ipi-openstack.poc.com:6443/version": dial tcp 10.157.36.21:6443: i/o timeout

DEBUG Still waiting for the Kubernetes API: Get "https://api.ipi-openstack.poc.com:6443/version": dial tcp 10.157.36.21:6443: i/o timeout

ERROR Attempted to gather ClusterOperator status after installation failure: listing ClusterOperator objects: Get "https://api.ipi-openstack.poc.com:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 10.157.36.21:6443: i/o timeout

DEBUG Fetching Bootstrap SSH Key Pair...

DEBUG Loading Bootstrap SSH Key Pair...

DEBUG Using Bootstrap SSH Key Pair loaded from state file

DEBUG Reusing previously-fetched Bootstrap SSH Key Pair

DEBUG Fetching Install Config...

DEBUG Loading Install Config...

DEBUG   Loading SSH Key...

DEBUG   Loading Base Domain...

DEBUG     Loading Platform...

DEBUG   Loading Cluster Name...

DEBUG     Loading Base Domain...

DEBUG     Loading Platform...

DEBUG   Loading Networking...

DEBUG     Loading Platform...

DEBUG   Loading Pull Secret...

DEBUG   Loading Platform...

DEBUG Using Install Config loaded from state file

DEBUG Reusing previously-fetched Install Config

INFO Pulling debug logs from the bootstrap machine

DEBUG failed to parse SSH private key from "/root/.ssh/cloud.key.pub"

DEBUG failed to parse SSH private key from "/root/.ssh/id_rsa.pub"

DEBUG failed to parse SSH private key from "/root/.ssh/known_hosts"

DEBUG failed to parse SSH private key from "/root/.ssh/undercloud.crt"

DEBUG Added /tmp/bootstrap-ssh1595526000 to installer's internal agent

DEBUG Added /root/.ssh/cloud.key to installer's internal agent

DEBUG Added /root/.ssh/id_rsa to installer's internal agent

ERROR Attempted to gather debug logs after installation failure: failed to create SSH client: dial tcp 10.157.36.10:22: connect: connection timed out

ERROR Bootstrap failed to complete: Get "https://api.ipi-openstack.poc.com:6443/version": dial tcp 10.157.36.21:6443: i/o timeout

ERROR Failed waiting for Kubernetes API. This error usually happens when there is a problem on the bootstrap host that prevents creating a temporary control plane.

ERROR Attempted to analyze the debug logs after installation failure: could not open the gather bundle: open : no such file or directory

FATAL Bootstrap failed to complete

Pls guide to resolve the issue.

**Comment by vrutkovs** [comment]
See https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting

**Reply by giteshchavan09** [reply]
Thanks for replying but we already tried the same. VC but no luck

**Reply by giteshchavan009** [reply]
Any help???  still suffering from same issue

**Comment by biruand1016** [comment]
I have a same issue. How can I solve this issue?
